src/ai_commands.js

```diff
@@ -4,45 +4,97 @@ import path from 'path';
+const config = Object.assign({}, workspace.getConfiguration('autoit'));
 
-// Executable paths
-const { aiPath, wrapperPath, tidyPath, checkPath, helpPath, infoPath, kodaPath } = configuration;
+workspace.onDidChangeConfiguration(({affectsConfiguration}) => {
+  if (!affectsConfiguration("autoit"))
+    return;
 
-const aiOutAll = window.createOutputChannel('AutoIt', 'vscode-autoit-output');
+  Object.assign(config, workspace.getConfiguration('autoit'));
+});
 
-// Additional help files
-const { smartHelp } = configuration;
+
+const aiOutAll = window.createOutputChannel('AutoIt', 'vscode-autoit-output');
 
 const runners = {
-  current: null,
-  list: new Map(),
-  isNewLine: true,
-  lastId: 0,
-  id: 1
-};
+  list: new Map(), //list of running scripts
+  aiOut: new Map(), //list of existed script outputs
+  isNewLine: true, //was previous message ended with a newline?
+  aiOutMaxCount: 2, //max number of aiOutProcess outputs to keep after script exited
+  aiOutTimeout: 60000, //timeout in msec to automatically close output after script exited
+  lastId: 0, //last output id printed text (used to show ID or not)
+  id: 0,
+  get lastRunning()
+  {
+    return this.lastRunningScript();
+  },
+  get lastRunningOpened()
+  {
+    return this.lastRunningScript(getActiveDocumentFile());
+  },
 
-const getLastRunningScript = () =>
-{
-  const thisFile = window.activeTextEditor.document.fileName;
-  for(let list = [...runners.list.entries()], i = list.length-1, runner, info; i >= 0; i--)
+  findLastAiOut(thisFile, processCommand)
   {
-    [runner, info] = list[i];
-    if (info.thisFile == thisFile)
+    for(let list = [...this.aiOut.entries()], i = list.length-1, runner, info; i >= 0; i--)
     {
-      return {runner, info};
+      [runner, info] = list[i];
+      if (info.thisFile == thisFile && info.processCommand == processCommand)
+      {
+        return {runner, info};
+      }
+    }
+    return null;
+  },
+  lastRunningScript(thisFile = null)
+  {
+    for(let list = [...this.list.entries()], i = list.length-1, runner, info; i >= 0; i--)
+    {
+      [runner, info] = list[i];
+      if (!thisFile || info.thisFile == thisFile)
+      {
+        return {runner, info};
+      }
+    }
+    return null;
+  },
+  aiOutCleanup({id, thisFile, runner, aiOutProcess, processCommand} = {})
+  {
+    const that = this;
+    const callback = () =>
+    {
+      aiOutProcess.dispose();
+      if (that.id == id)
+        aiOutAll.show(true); //switch to main output
+
+      that.aiOut.delete(runner);
+    };
+    if (runner)
+      this.aiOut.set(runner, {timer: this.aiOutTimeout ? setTimeout(callback, this.aiOutTimeout) : null, callback, thisFile, processCommand, aiOutProcess, id});
+
+    while (this.aiOut.size > this.aiOutMaxCount)
+    {
+      const {timer, callback} = this.aiOut.values().next().value;
+      clearTimeout(timer);
+      callback();
     }
   }
-  return null;
 };
 
+function getActiveDocumentFile() {
```

Comment:
`window.activeTextEditor.document.fileName` is not available in some situations (like when `runScript()` executed settings tab), so instead of displaying error, we simply return empty string.